dashboard/serializers.py

```diff
@@ -0,0 +1,52 @@
+"""
+Provides functionality for serializing a ProgramEnrollment for the ES index
+"""
+from decimal import Decimal
+
+
+class UserProgramSerializer:
+    """
+    Provides functions for serializing a ProgramEnrollment for the ES index
+    """
+    @staticmethod
+    def calculate_certificate_grade_average(certificates):
+        """
+        Calculates an average grade (integer) from a list of certificates with <1 decimal grades
+        """
+        return None if len(certificates) == 0 else round(
+            (sum([Decimal(certificate['grade']) for certificate in certificates]) / len(certificates)) * 100
+        )
+
+    @classmethod
+    def serialize_valid_edx_data(cls, related_edx_obj_set, program):
+        """
+
+        Args:
+            related_edx_obj_set: Related QuerySet of cached edX model objects (eg: CachedEnrollment)
```

Comment:
The types within the parentheses are read by PyCharm and possibly other editors, so it's useful to include them even if they read as redundant in English

The edX guide: http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python_guidelines.html#docstrings

I'm fine with just

    Returns:
        list of dict: List of serialized objects

There isn't really a standard for generic types in sphinx-napoleon docstrings AFAIK so it seems like the best we can do here. For 1 I'm fine with `RelatedManager` there, it would let the editors infer certain manager functions at least